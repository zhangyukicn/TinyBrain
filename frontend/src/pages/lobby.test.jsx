import React from 'react';
import { render, screen } from '@testing-library/react';
import { configure, EnzymeAdapter, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Lobby from './lobby';
configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
      pathname: './lobby',
      state: { id: 173450416 }
    }),
  }));

  describe('Lobby', () => {
  const modalStart = shallow(<Lobby />);

  it('is matched with snapshot', () => {
    expect(modalStart).toMatchSnapshot();
  });

  it('can display correct content when I enter Lobby', () => {
    expect(modalStart.find('h1').text()).toEqual('Waiting for game beginning!');
  });
})
