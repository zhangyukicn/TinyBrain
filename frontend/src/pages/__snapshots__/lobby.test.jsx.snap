// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby is matched with snapshot 1`] = `ShallowWrapper {}`;
